index.js

```diff
@@ -135,6 +137,45 @@ async function run() {
           containerDef.environment.push(variable);
         }
       })
+
+      if (secrets) {
+        // If secrets array is missing, create it
+        if (!Array.isArray(containerDef.secrets)) {
+          containerDef.secrets = [];
+        }
+
+        // Get pairs by splitting on newlines
+        secrets.split('\n').forEach(function (line) {
+          // Trim whitespace
+          const trimmedLine = line.trim();
+          // Skip if empty
+          if (trimmedLine.length === 0) { return; }
+          // Split on =
+          const separatorIdx = trimmedLine.indexOf("=");
+          // If there's nowhere to split
+          if (separatorIdx === -1) {
+              throw new Error(
+                `Cannot parse the secret '${trimmedLine}'. Secret pairs must be of the form NAME=valueFrom, 
+                where valueFrom is an arn from parameter store or secrets manager. See AWS documentation for more information: 
+                https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specifying-sensitive-data.html.`);
+          }
```

Comment:
These lines are identical with the lines 113 to 122 above. We could extract and create a common method that does this functionality of trimming and splitting?